apps/frontend/src/modules/workspace/hooks/useWorkspace.tsx

```diff
@@ -58,14 +60,54 @@ export const useWorkspace = () => {
                         },
                     },
                 });
-                return response?.data?.organizationUpdate;
+                if (response?.data?.organizationUpdate) {
+                    const updatedUser = JSON.parse(JSON.stringify(user)) as DeepPartial<User>;
+                    const currentOrganization = updatedUser.organizations?.edges?.find(
+                        (org) => org?.node?.id === workspace?.id,
+                    );
+                    if (currentOrganization) {
+                        currentOrganization.node = {
+                            ...currentOrganization.node,
+                            name: organizationInput.name || "",
+                            slug: organizationInput.slug || "",
+                        };
+                    }
+                    updateUser(updatedUser, true);
+                    toast.success(`Your organization name has been updated`);
+                    return;
+                }
             } catch (error) {
                 console.error(error);
             }
         },
         [updateUser, user],
     );
 
+    const handleWorkspaceCache = useCallback(
```

Comment:
@afolasope this function updates the workspace cache, can it be merged with the `updateWorkspace` function so that when it's called with `cacheOnly` set to `false`, it makes a network request and updates the cache and when it is `true` it updates only the cache?